lucene/core/src/java/org/apache/lucene/search/FuzzyQuery.java

```diff
@@ -237,22 +216,9 @@ public boolean equals(Object obj) {
     if (getClass() != obj.getClass())
       return false;
     FuzzyQuery other = (FuzzyQuery) obj;
-    // Note that we don't need to compare termLength or automata because they
-    // are entirely determined by the other fields
-    if (maxEdits != other.maxEdits)
-      return false;
-    if (prefixLength != other.prefixLength)
-      return false;
-    if (maxExpansions != other.maxExpansions)
-      return false;
-    if (transpositions != other.transpositions)
-      return false;
-    if (term == null) {
-      if (other.term != null)
-        return false;
-    } else if (!term.equals(other.term))
-      return false;
-    return true;
+    return Objects.equals(maxEdits, other.maxEdits) && Objects.equals(prefixLength, other.prefixLength)
```

Comment:
nitpick: compare the term first, as it is most likely to return false for different fuzzy